webClient/src/app/services/Uploader.service.ts

```diff
@@ -1,94 +1,105 @@
 import { Injectable } from '@angular/core';
 import { HttpClient, HttpParams } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
 
 @Injectable()
 export class UploaderService {
     constructor(private http: HttpClient) { }
 
-    chunkAndSendFile(file: File, uploadDirPath: string): void {
-        const fileSize = file.size;
-        const chunkSize =  3 * 1024 * 1024; // bytes
-        let sourceEncoding = 'BINARY';
-        let targetEncoding = 'BINARY';
-        let chunkIdx = 0;
-        let offset = 0;
-        let sessionID: number;
-        let uri = ZoweZLUX.uriBroker.unixFileUri('contents', uploadDirPath.slice(1) + '/' + file.name, sourceEncoding, targetEncoding, undefined, true);
+    // uploadDirPath should be a directory on the remote server without a / at the end.
+    // Example: '/u/ts6531'
+    chunkAndSendFile(file: File, uploadDirPath: string, targetEncoding: string): Observable<number> {
+        return new Observable(
+            observer => {
+                const fileSize = file.size;
+                const chunkSize =  3 * 1024 * 1024; // bytes
+                let sourceEncoding;
+                targetEncoding === 'BINARY' ? sourceEncoding = 'BINARY' : sourceEncoding = 'UTF-8';
+                let chunkIdx = 0;
+                let offset = 0;
+                let sessionID: number;
+                const uri = ZoweZLUX.uriBroker.unixFileUri('contents', uploadDirPath.slice(1) + '/' + file.name,
+                sourceEncoding, targetEncoding, undefined, true);
 
-        console.table({'URI': uri, 'File Name': file.name, 'File Size': fileSize, 'Chunk Size': chunkSize});
+                console.table({'URI': uri, 'File Name': file.name, 'File Size': fileSize, 'Chunk Size': chunkSize});
```

Comment:
I would say using console.table here is permitted since we don't have such a feature in the logger yet. Sometime down the road perhaps.